packages/adapter-vercel/index.js

```diff
@@ -95,22 +98,24 @@ const plugin = function ({ external = [], edge, split } = {}) {
 				write(
 					`${dirs.functions}/${name}.func/.vc-config.json`,
 					JSON.stringify({
+						...config,
 						runtime: 'edge',
 						entrypoint: 'index.js'
-						// TODO expose envVarsInUse
 					})
 				);
 
-				config.routes.push({ src: pattern, dest: `/${name}` });
+				static_config.routes.push({ src: pattern, dest: `/${name}` });
 			}
 
-			const generate_function = edge ? generate_edge_function : generate_serverless_function;
-
-			if (split) {
+			if (split || builder.hasRouteLevelConfig) {
 				await builder.createEntries((route) => {
+					const route_config = { ...default_config, ...route.config };
 					return {
 						id: route.pattern.toString(), // TODO is `id` necessary?
-						filter: (other) => route.pattern.toString() === other.pattern.toString(),
+						filter: (other) =>
+							split
+								? route.pattern.toString() === other.pattern.toString()
+								: can_group(route_config, { ...default_config, ...other.config }),
```

Comment:
There's a subtle issue here — suppose you had two routes, `/[a=one]` and `/[b=two]`, and they had incompatible configuration.

In the `split: true` case, they would be incorrectly bundled into the same function because they would have the same pattern.

In the `split: false` case, they would be bundled separately, but because of the matching patterns only one function would ever run; requests that belong to the second route would result in a 404.

There's no way for the adapter to solve this, but we _could_ identify these cases and throw a 'change your config or alter your routes' error.